### ISSUE TITLE
SSL option not working

### ISSUE THREAD

**Issue by mrminecart** [Grokzen/redis-py-cluster#252]
Passing ssl option does not seem to make a difference when trying a connection

```
>>> rc = StrictRedisCluster(startup_nodes=[{"host": "host", "port": 6379}], decode_responses=True, ssl=True, password='password')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/site-packages/rediscluster/client.py", line 181, in __init__
    **kwargs
  File "/usr/lib/python2.7/site-packages/rediscluster/connection.py", line 141, in __init__
    self.nodes.initialize()
  File "/usr/lib/python2.7/site-packages/rediscluster/nodemanager.py", line 183, in initialize
    raise RedisClusterException("ERROR sending 'cluster slots' command to redis server: {0}".format(node))
rediscluster.exceptions.RedisClusterException: ERROR sending 'cluster slots' command to redis server: {'host': 'host', 'port': 6379}
```

I am using version: `redis-py-cluster (1.3.4)`

**Comment by Grokzen** [comment]
Ye i think i know what this is, the initialize() method sends some probing commands to the cluster to determine the cluster setup before initialize the client data structures. It should be a rather easy fix to solve by making all connections using the same methods instead of how it is now.

**Comment by Grokzen** [comment]
This problem could have been solved by this commit b12c3538b3b1c46e7ba7ce9f78c245a80ced5f7f

It appears that the correct SSL connection class was not set when you are using `ssl=True`

**Comment by akrylysov** [comment]
https://github.com/Grokzen/redis-py-cluster/commit/b12c3538b3b1c46e7ba7ce9f78c245a80ced5f7f fixed the issue, thanks!
